core/src/Kokkos_Macros.hpp

```diff
@@ -58,6 +58,18 @@
 #include <impl/Kokkos_NvidiaGpuArchitectures.hpp>
 #endif
 
+#if !defined(KOKKOS_ENABLE_CXX17)
+#if __has_include(<version>)
+#include <version>
+#else
+#include <iso646.h>
```

Comment:
```suggestion
#include <ciso646>
```

Reply:
https://en.cppreference.com/w/cpp/header/ciso646 says:
> [...] \<ciso646> is removed in C++20. Corresponding <iso646.h> is still available in C++20. [...]

so it seems safer to use `iso646.h`.

Reply:
The whole thing is guarded for C++20 enabled